autoscaling.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	log "github.com/sirupsen/logrus"
+)
+
+// Envelope ...
+type Envelope struct {
+	Type    string    `json:"Type"`
+	Subject string    `json:"Subject"`
+	Time    time.Time `json:"Time"`
+	Message string    `json:"Message"`
+}
+
+// Message ...
+type Message struct {
+	Time        time.Time `json:"Time"`
+	GroupName   string    `json:"AutoScalingGroupName"`
+	InstanceID  string    `json:"EC2InstanceId"`
+	ActionToken string    `json:"LifecycleActionToken"`
+	Transition  string    `json:"LifecycleTransition"`
+	HookName    string    `json:"LifecycleHookName"`
+}
+
+// NewAutoscalingListener ...
+func NewAutoscalingListener(instanceID string, queue *Queue, autoscaling *autoscaling.AutoScaling) *AutoscalingListener {
```

Comment:
@lox - as mentioned in your comment, we could rename this file and all the structs from `Autoscaling` to:
- `lifecycle.go`
- `LifecycleListener`
- `lifecycleTerminationNotice`

`Type()` should return `"lifecycle"` as well then?

Reply:
I'm fine with leaving it with `Autoscaling`.